dallinger/recruiters.py

```diff
@@ -96,6 +96,12 @@ def notify_duration_exceeded(self, participants, reference_time):
             "have been active for too long. No action taken."
         )
 
+    def ping(self):
+        """Called on a schedule, in case any action needs to be taken based on
+        the current state of recruitment.
+        """
+        pass
```

Comment:
Was this method a work in progress?

Reply:
No, the base class provides an implementation that does nothing. Some recruiters (like the PulseRecruiter) will need to perform some check and potentially take action when called. Having a no-op default means the clock code can just call the recruiter without checking what recruiter it is, and whether it supports `ping()`.